modules/mesh-task/variables.tf

```diff
@@ -477,8 +441,36 @@ variable "consul_ecs_config" {
 
 }
 
-variable "audit_logging" {
-  description = "Whether to enable audit logging for the Consul agent [Consul Enterprise]. ACLs must be enabled to enable audit logging."
-  type        = bool
-  default     = false
+variable "http_tls_config" {
+  type        = any
+  default     = {}
+  description = <<-EOT
+  This accepts HTTP specific TLS configuration based on the `consulServers.http` schema present in https://github.com/hashicorp/consul-ecs/blob/main/config/schema.json.
+  If empty, values of `var.tls`, `var.tls_server_name` and `var.ca_cert_file` will be used to configure TLS settings for HTTP. 
+  EOT
+
+  validation {
+    error_message = "Only the 'port', 'https', 'tls', 'tlsServerName' and 'caCertFile' fields are allowed in http_tls_config."
+    condition = alltrue([
+      for key in keys(var.http_tls_config) :
+      contains(["port", "https", "tls", "tlsServerName", "caCertFile"], key)
+    ])
+  }
 }
+
+variable "grpc_tls_config" {
```

Comment:
Exposing first class variables for every field in gRPC and http tls settings seem like an overkill. Exposing them as first class config variables which we will merge with the default setting in `config.tf`. Open for feedback on this.